DOCSP-13854 usage examples landing page (#22)
mongodb/docs-golang: source/usage-examples.txt

--------

Usage examples provide convenient starting points for popular MongoDB
operations. Each example provides the following:

> - A full Go program that to run in your own environment

Reviewer: Change: Typo.
```suggestion
- A full Go program that you can run in your own environment
```

Atlas by following the :atlas:`Get Started with Atlas Guide
<getting-started/#atlas-getting-started>` or you can
:guides:`import the sample dataset into a local MongoDB instance
</server/import/>`.

Once you import the dataset, you can copy and paste a usage
example into your development environment of choice. You can follow the
:doc:`quick start guide </quick-start>` to learn more about getting
started with the MongoDB Go Driver. 

To connect the example to your MongoDB instance, you must :ref:`define
an environment variable <usage-examples-env-variable>` with
:ref:`your connection string <usage-examples-connection-string>`. 

.. code-block:: go
   
   uri := os.Getenv("MONGODB_URI")

.. _usage-examples-env-variable:

Environment Variable
~~~~~~~~~~~~~~~~~~~~

You can use `GoDotEnv <https://github.com/joho/godotenv>`__ to set up
your environment variable. 

Add the following application configuration in your ``.env`` file at the
root of your project. For more information, see the
`GoDotEnv documentation <https://github.com/joho/godotenv#usage>`__.

.. code-block::

   MONGODB_URI=mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority

.. note::

>    Replace the preceding with your MongoDB deployment's connection string.

Reviewer: Nit: The preceding what? Maybe "the preceding example connection string"?

> Once you have imported the dataset, you can copy and paste a usage

Reviewer: Change: Simplest tense.
```suggestion
Once you import the dataset, you can copy and paste a usage
```

example into your development environment of choice. You can follow the
:doc:`quick start guide </quick-start>` to learn more about getting
started with the MongoDB Go Driver. 

> Once you've copied a usage example, you'll need to edit and place
> :ref:`your connection string <usage-examples-connection-string>` in
> :ref:`your environment variable <usage-examples-env-variable>` to get
> the example connected to your instance of MongoDB. 

Reviewer: Change: 
- Use the simplest tense. [style guide ref](https://docs.mongodb.com/meta/style-guide/writing/use-effective-verbs/#use-the-simplest-tense)
- I think its understood that they will have copied and pasted (especially given the previous paragraph) and we can simplify this down.

```suggestion
To connect the example to your instance of MongoDB, you must edit
:ref:`your environment variable <usage-examples-env-variable>` to include 
:ref:`your connection string <usage-examples-connection-string>`. 
```

.. code-block:: go
   
   uri := os.Getenv("MONGODB_URI")

.. _usage-examples-env-variable:

Environment Variable
~~~~~~~~~~~~~~~~~~~~

You can use `GoDotEnv <https://github.com/joho/godotenv>`__ to set up
your environment variable. 

> When you get to the "Usage" section, add the following application
> configuration in your ``.env`` file at the root of your project: 

Reviewer: Change: At first I didn't know which Usage section you were talking about. I think I would include this as additional context instead.

```suggestion
Add the following application configuration in your ``.env`` 
file at the root of your project. For more information, see the
`GoDotEnv documentation <https://github.com/joho/godotenv#usage>`__.
```

.. code-block::

   MONGODB_URI=mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority

.. note::

>    Replace the following with your MongoDB deployment's connection string.

Reviewer: Question: There is nothing following... was this copypasta?

Author: 😅 I rearranged it and forgot to update it to say "preceding"

.. _usage-examples-connection-string:

Connection String
~~~~~~~~~~~~~~~~~

You can use the :guides:`Atlas Connectivity Guide </cloud/connectionstring/>`
to enable connectivity to your instance of Atlas and find the
:manual:`connection string </reference/connection-string/>` to set the
``MONGODB_URI`` environment variable in the usage example. If your instance uses
:manual:`SCRAM authentication </core/security-scram/>`, you can replace
``<user>`` with your username, ``<password>`` with your password, and
``<cluster-url>`` with the IP address or URL of your instance.

For more information about connecting to your MongoDB instance, see the

> <TODO: Connection Guide>.

Reviewer: Question: Just making sure you knew this TODO was here? Don't want it to go out in prod :)

Author: Yes, there's quite a few throughout at the moment -- we'll be doing a pass to update them before putting the docs in prod :)
